DigitalLearningSolutions.Web/Controllers/LearningPortalController/RecommendedLearningController.cs

```diff
@@ -0,0 +1,183 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.LearningPortalController
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.External.Filtered;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Helpers.ExternalApis;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.LearningPortal.SelfAssessments.FilteredMgp;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Configuration;
+
+    [Authorize(Policy = CustomPolicies.UserOnly)]
+    [ServiceFilter(typeof(VerifyDelegateUserCanAccessSelfAssessment))]
+    public class RecommendedLearningController : Controller
+    {
+        private readonly IConfiguration configuration;
+        private readonly IFilteredApiHelperService filteredApiHelperService;
+        private readonly ISelfAssessmentService selfAssessmentService;
+
+        public RecommendedLearningController(
+            IFilteredApiHelperService filteredApiHelperService,
+            ISelfAssessmentService selfAssessmentService,
+            IConfiguration configuration
+        )
+        {
+            this.filteredApiHelperService = filteredApiHelperService;
+            this.selfAssessmentService = selfAssessmentService;
+            this.configuration = configuration;
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/Results")]

```

Comment:
There's an action SetCompleteAsset that doesn't have a route that we'd probably need to verify I haven't broken by adding a route to it.

Reply:
Wouldn't the route have already broken with this moving into a new controller?

Reply:
It should work fine at the default value (as it did before). Adding the Route to the controller would break it unless we also added some route to the action as well.